### PR TITLE
feat: multiwindow support; render

### PR DESCRIPTION
**Pull request by apollo1321**

### CODE REVIEW

File: lua/treesitter-context/render.lua [hunk 1]
```diff
@@ -6,49 +6,43 @@ local config = require('treesitter-context.config')
 
 local ns = api.nvim_create_namespace('nvim-treesitter-context')
 
--- Don't access directly, use get_bufs()
-local gutter_bufnr --- @type integer?
-local context_bufnr --- @type integer?
+--- @class ContextIDs
```
Comment (lewis6991) [inline lua/treesitter-context/render.lua:9]:
```suggestion
--- @class ContextID
```

File: lua/treesitter-context/render.lua [hunk 2]
```diff
@@ -6,49 +6,43 @@ local config = require('treesitter-context.config')
...
+--- @field winid integer
+--- @field bufnr integer
```
[the remark this answers is not in the document]
Comment (apollo1321) [reply lua/treesitter-context/render.lua:11]: I'm not sure how necessary it is. I wanted to save a couple of `nvim_win_get_buf` calls, but the code looks cleaner without carrying the `bufnr`.

[the remark this answers is not in the document]
Comment (lewis6991) [reply lua/treesitter-context/render.lua:11]: Let's try without carrying `bufnr` then.

File: lua/treesitter-context/render.lua [hunk 3]
```diff
@@ -381,33 +376,42 @@ function M.open(bufnr, winid, ctx_ranges, ctx_lines)
 
   local win_height = math.max(1, #ctx_lines)
 
-  local gbufnr, ctx_bufnr = get_bufs()
+  local window_context = window_contexts[winid] or {
+    context_winid = nil,
+    gutter_winid = nil,
+  }
```
Comment (lewis6991) [inline lua/treesitter-context/render.lua:382]:
```suggestion
  local window_context = window_contexts[winid] or {}
```

File: lua/treesitter-context/render.lua [hunk 4]
```diff
@@ -381,33 +376,42 @@ function M.open(bufnr, winid, ctx_ranges, ctx_lines)
...
+  if window_contexts[winid] == nil then
+    window_contexts[winid] = window_context
+  end
```
Comment (lewis6991) [inline lua/treesitter-context/render.lua:385]:
```suggestion
  window_contexts[winid] = window_contexts[winid] or {}
  local window_context = window_contexts[winid] 
```

**Comment by apollo1321** [comment]
Ping

**Pull request merged**